lib/ansible/modules/cloud/amazon/kinesis_firehose.py

```diff
@@ -0,0 +1,626 @@
+            if key[0] == '_':
+                key = key[1:]
+            if isinstance(val, datetime.datetime):
+                results[key] = val.isoformat()
+            elif isinstance(val, dict):
+                results[key] = convert_to_snake_case(val)
+            elif isinstance(val, list):
+                converted = list()
+                for item in val:
+                    converted.append(convert_to_snake_case(item))
+                results[key] = converted
+            else:
+                results[key] = val
+    return results
+
+
+def convert_to_title_case(data):
+    """Convert all snake_case keys in dict with TitleCase
+    Args:
+        data (dict): Dictionary with keys that have snake_case in them
+            Example.. foo_bar == FooBar
+            if a val is of type datetime.datetime, it will be converted to
+            the ISO 8601
+
+    Basic Usage:
+        >>> test = {'foo_bar': []}
+        >>> test = convert_to_title_case(test)
+        {
+            'FooBar': []
+        }
+
+    Returns:
+        Dictionary
+    """
+    results = dict()
+    if isinstance(data, dict):
+        for key, val in data.items():
+            key = to_title(key)
+            if key[0] == '_':
+                key = key[1:]
+            if isinstance(val, datetime.datetime):
+                results[key] = val.isoformat()
+            elif isinstance(val, dict):
+                results[key] = convert_to_title_case(val)
+            elif isinstance(val, list):
+                converted = list()
+                for item in val:
+                    converted.append(convert_to_title_case(item))
+                results[key] = converted
+            else:
+                results[key] = val
+    return results
+
+
+def find_stream(client, delivery_stream_name, check_mode=False):
+    """Retrieve a Kinesis Firehose Delivery Stream.
+    Args:
+        client (botocore.client.EC2): Boto3 client.
+        stream_name (str): Name of the Kinesis Firehose Delivery stream.
+
+    Kwargs:
+        check_mode (bool): This will pass DryRun as one of the parameters to the aws api.
+            default=False
+
+    Basic Usage:
+        >>> client = boto3.client('firehose')
+        >>> stream_name = 'test-stream'
+
+    Returns:
+        Tuple (bool, str, dict)
+    """
+    err_msg = ''
+    success = False
+    params = {
+        'DeliveryStreamName': delivery_stream_name,
+    }
+    results = dict()
+    try:
+        if not check_mode:
+            results = (
+                client.describe_delivery_stream(**params)['DeliveryStreamDescription']
+            )
+        else:
+            results = {
+                'StreamName': delivery_stream_name,
+                'StreamARN': 'arn:aws:kinesis:east-side:123456789:stream/{0}'.format(delivery_stream_name),
+                'DeliveryStreamStatus': 'ACTIVE'
+            }
+        success = True
+    except botocore.exceptions.ClientError as e:
+        err_msg = to_native(e)
```

Comment:
You should follow standard [Ansible AWS Exception handling guidelines](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#exception-handling-for-boto3-and-botocore)